Obsidian/WorldData/World.cs

```diff
@@ -24,42 +31,44 @@ public class World : IWorld
     public ConcurrentBag<(int X, int Z)> SpawnChunks { get; private set; } = new();
 
     public string Name { get; }
+    public string Seed { get; }
+    public string FolderPath { get; private set; }
+    public string PlayerDataPath { get; private set; }
+    public string LevelDataFilePath { get; private set; }
 
     public bool Loaded { get; private set; }
 
-    public long Time => Data.Time;
+    public long Time => LevelData.Time;
 
-    public Gamemode GameType => Data.GameType;
+    public Gamemode DefaultGamemode => LevelData.DefaultGamemode;
 
-    private float rainLevel = 0f;
+    public string DimensionName { get; private set; }
 
-    internal World(string name, Server server)
-    {
-        this.Data = new Level
-        {
-            Time = 1200,
-            GameType = (int)Gamemode.Survival,
-            GeneratorName = WorldType.Default.ToString()
-        };
+    public string? ParentWorldName { get; private set; }
 
+    internal World(string name, Server server, string seed, Type generatorType)
+    {
         this.Name = name ?? throw new ArgumentNullException(nameof(name));
         this.Server = server;
 
-        var playerDataPath = Path.Combine(this.Server.ServerFolderPath, this.Name, "playerdata");
-        if (!Directory.Exists(playerDataPath))
-            Directory.CreateDirectory(playerDataPath);
+        this.Seed = seed ?? throw new ArgumentNullException(nameof(seed));
+
+        this.Generator = (IWorldGenerator)Activator.CreateInstance(generatorType);
+        this.Generator.Init(seed);
     }
 
     public int GetTotalLoadedEntities() => this.Regions.Values.Sum(e => e == null ? 0 : e.Entities.Count);
 
-    public async Task UpdateClientChunksAsync(Client c, bool unloadAll = false)
+    public async Task UpdateClientChunksAsync(Client c, bool unloadAll = false, bool respawning = false)
```

Comment:
Is `respawning` ever true?
Should this be a property of Client?

Reply:
YOu're right I actually didn't think about that I'll make it a property 🙏🏼